### ISSUE TITLE
Resolving API token, please wait...

### ISSUE THREAD

**Issue by macfreek** [dennisreimann/ioctocat#168]
Hi, 3 days ago, I wanted to use Ioctocat again (after not really using it for some months).
When I start the app, I first get a message "Authenticating, please wait...", followed after 1-2 seconds by "Resolving API token, please wait...". This last message does not go away, even not after a few minutes.

I have this behaviour on my iPhone 3GS with iOS 5 and iOctocat 1.5._, as well as on my iPhone 5 with iOS6 and both iOctocat 1.5._ and 1.6.

I tried to debug the traffic on my router, but it all seems port 443 (thus encrypted). For what it's worth, I see connections to and replies from:
- 207.97.227.243 - gist.github.com.
- 207.97.227.239 - github.com.
- 72.233.61.123 - gravatar.com
- 2a00:1450:4002:802::101e - *.google-analytics.com
- 64.22.108.249 - get.gaug.es (some traffic analysis?)

I can log in on http://www.github.com just fine (otherwise, I wouldn't be able to leave this message ;-) )

Any clue how to further debug this?

**Comment by dennisreimann** [comment]
Can you try to restart the app? Unfortunately there is only one API call left for which one needs the token (that's the organization feeds). I will get rid of that as soon as possible, which will make this obsolete.

**Comment by macfreek** [comment]
If I simply bring the app back to the foreground, I immediately see the "Resolving API token" message again.

If I quit the app (press home button twice, swipe to find iOctocat, then press and hold till the minus -quit- mark appears), then I get the same problem -- first the "Authenticating" message, than the "Resolving API token" again.

Would implementing OAuth (#150) resolve this? In that case feel free to close this report as "won't fix" and instead work on that. I wouldn't mind waiting a month or so for that.

**Comment by dennisreimann** [comment]
For now you can circumvent it by going to the account settings (use the blue arrow on the right of the account row) and  putting 32 characters in the token field. Or, if you want that token to work (like I said you'll need it for the org feeds), go to your GitHub dashboard, copy the link location of the feed, extract the token param and put the value into the token field in the settings. I am very sorry for this, I know this terribly sucks - but that's what the app is trying to do in the background and it seems to get stuck somewhere during that process.

I'll fix this with the upcoming release, promised!

**Comment by macfreek** [comment]
That was a bit of a pickle. As soon as I start iOctocat, it tries to authenticate, and while it's waiting indefinitely to do so, it greys out all buttons -- in short, I can't seem to change any settings.

I did find a workaround -- delete the iOctocat app, and re-install it again from the App store. That way all settings were gone, and I could add them anew. I got the token from the RSS/Atom feed, and used that.

To good news -- it works now, thanks!
